python/dolfinx/mesh.py

```diff
@@ -62,13 +63,82 @@
     "to_string",
     "transfer_meshtag",
     "entities_to_geometry",
+    "create_geometry",
+    "Geometry",
 ]
 
 
+class Geometry:
+    """The geometry of a :class:`dolfinx.mesh.Mesh`"""
+
+    _cpp_object: typing.Union[_cpp.mesh.Geometry_float32, _cpp.mesh.Geometry_float64]
+
+    def __init__(
+        self, geometry: typing.Union[_cpp.mesh.Geometry_float32, _cpp.mesh.Geometry_float64]
+    ):
+        """Initialize a geometry from a C++ geometry.
+
+        Args: The C++ geometry object.
+
+        Note:
+            Geometry objects should usually be constructed with the :func:`create_geometry`
+            and not this class initializer. This class is combined with different base classes
+            that depend on the scalar type used in the Geometry.
+        """
+
+        self._cpp_object = geometry
+
+    @property
+    def cmap(self) -> _CoordinateElement:
+        """The element that describes the geometry map."""
+        return _CoordinateElement(self._cpp_object.cmap)
+
+    def cmaps(self, i: int) -> _CoordinateElement:
```

Comment:
I don't think we need this yet. It is highly experimental. Best to remove for now.